packit_service/worker/jobs.py

```diff
@@ -710,38 +717,60 @@ def get_config_for_handler_kls(
 
         return matching_jobs
 
-    def push_initial_metrics(
+    def push_statuses_metrics(
+        self,
+        statuses_check_feedback: List[datetime],
+    ):
+        """
+        Push the metrics about the time of setting initial statuses for the first and last check.
+
+        Args:
+            statuses_check_feedback: A list of times it takes to set every initial status check.
+        """
+        if statuses_check_feedback:
+            pushgateway = Pushgateway()
+            response_time = elapsed_seconds(
+                begin=self.event.created_at, end=statuses_check_feedback[0]
+            )
+            logger.debug(
+                f"Reporting first initial status check time: {response_time} seconds."
+            )
+            pushgateway.first_initial_status_time.observe(response_time)
+            if response_time > 30:
+                pushgateway.no_status_after_30_s.inc()
+                # https://github.com/packit/packit-service/issues/1728
+                # we need more info why this has happened
+                logger.debug(f"Event dict: {self.event}.")
+                logger.error(
+                    f"Event {self.event.__class__.__name__} took ({response_time}s) to process."
+                )
+            # set the time when the accepted status was set so that we
+            # can use it later for measurements
+            self.event.task_accepted_time = statuses_check_feedback[0]
+
+            response_time = elapsed_seconds(
+                begin=self.event.created_at, end=statuses_check_feedback[-1]
```

Comment:
Why do you take the last one here?

Oh, I see, it's the last one… Are we sure at this point that all statuses have been processed?

Reply:
I am quite sure they are, where by processed I mean all of them should have the very first status feedback set.